scripts/extensions/sams/src/utils/api.ts

```diff
@@ -30,9 +30,17 @@ export function getApiErrorMessage(error: IAPIError): string {
         return API_ERRORS[error.error](error);
     }
     if (error.errors?.name != null) {
```

Comment:
This will only respond to validation errors with the `name` field. (see https://superams.readthedocs.io/en/latest/errors/resource.html for a reference to this type of error).

It would also be better to move this error handling to the `API_ERRORS` dictionary under the `04001` error code, so that we can easily determine what error codes are handled by superdesk.

Reply:
The `error.errors` dictionary contains a mapping where the key is the name of the field that failed validation. Here `name` is the field that failed. This dictionary could contain other fields as well. So you would need to iterate over the keys as well